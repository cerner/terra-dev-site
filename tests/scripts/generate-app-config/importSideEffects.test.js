const importSideEffects = require('../../../scripts/generate-app-config/importSideEffects');
const ImportAggregator = require('../../../scripts/generate-app-config/generation-objects/ImportAggregator');

describe('importSideEffects', () => {
  it('finds the correct side effect js files', () => {
    const sideEffects = [
      '../tests/scripts/generate-app-config/sideEffectFiles/normal',
      '../tests/scripts/generate-app-config/sideEffectFiles/jsx',
      '../tests/scripts/generate-app-config/sideEffectFiles/extension.js',
      `${process.cwd()}/tests/scripts/generate-app-config/sideEffectFiles/fullpath`,
      '../tests/scripts/generate-app-config/sideEffectFiles/*/*.js',
      '../tests/scripts/generate-app-config/sideEffectFiles/strangeExt.derp',
    ];
    const imports = new ImportAggregator();
    importSideEffects(sideEffects, imports);
    expect(imports.toCodeString()).toMatchSnapshot();
  });
});
