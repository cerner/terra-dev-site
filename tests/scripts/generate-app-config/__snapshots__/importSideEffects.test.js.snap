// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importSideEffects finds the correct side effect js files 1`] = `
" import Component1 from '/Users/mh015940/source/terra-dev-site/tests/scripts/generate-app-config/sideEffectFiles/normal.js';
 import Component2 from '/Users/mh015940/source/terra-dev-site/tests/scripts/generate-app-config/sideEffectFiles/jsx.jsx';
 import Component3 from '/Users/mh015940/source/terra-dev-site/tests/scripts/generate-app-config/sideEffectFiles/extension.js';
 import Component4 from '/Users/mh015940/source/terra-dev-site/tests/scripts/generate-app-config/sideEffectFiles/fullpath.js';
 import Component5 from '/Users/mh015940/source/terra-dev-site/tests/scripts/generate-app-config/sideEffectFiles/glob/globa.js';
 import Component6 from '/Users/mh015940/source/terra-dev-site/tests/scripts/generate-app-config/sideEffectFiles/glob/globb.js';
 import Component7 from '/Users/mh015940/source/terra-dev-site/tests/scripts/generate-app-config/sideEffectFiles/strangeExt.derp.js';
"
`;
